### PR TITLE
Non weekend date year

### LINKED ISSUE #5223: Input validation regex too loose
```
export const ValidationTemplateInputType = {
  [TemplateInputType.TEXT]: `(.*)`,
  [ValidationType.WHOLE_NUMBER]: `[0-9]*`,
  [ValidationType.NUMBER]: `[0-9]+\.*[0-9]*`,
  [TemplateInputType.USER_DESCRIPTION_OUTCOME]: `(.*)`,
  [TemplateInputType.SUBSTITUTE_USER_OUTCOME]: `[0-9]*`,
  [TemplateInputType.DATETIME]: `(January|February|March|April|May|June|July|August|September|October|November|December) ([0-9]){2}, 20|([0-9]{2}) \d\d:\d\d (AM|PM) \\(UTC 0\\)`,
  [TemplateInputType.DATEYEAR]: `(January|February|March|April|May|June|July|August|September|October|November|December) ([0-9]){2}, 20|([0-9]{2})`,
};
```

These regex's are too loose. I was able to create a bunch fo markets that showed up as "Augur Templates", with strange values that get inserted. Here's some below

Screen Shot 2019-12-12 at 4 04 17 PM

It seems the regex's aren't always working correctly?  How did `4..1a` get through? Regardless of that, I think these would be better values:

```
  [ValidationType.NUMBER]: `[0-9]+(\.[0-9]+)?`,
  [TemplateInputType.SUBSTITUTE_USER_OUTCOME]: `[0-9]+`,

```

### PR DESCRIPTION
**Pull request by bthaile**
dont allow user to create template finance market where date picked is on a weekend. <https://github.com/AugurProject/augur/issues/5300>
also don't allow user to pick start date after end date in a time range finance template market.
created template validation tests, to start adding some examples of template validations
removed list values for indexes, also associated with <https://github.com/AugurProject/augur/issues/5300>

next effort will be for already created template market validations. (shield icon)

fix <https://github.com/AugurProject/augur/issues/5223>

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://dashboard.featurepeek.com). 🆒
Your deployment will be kept up-to-date with this pull request's latest changes.

**https://peek.run/654hjjz**

-------

*Please read [our docs](https://docs.featurepeek.com/configuring-peek-yml/) for more configuration details.*

File: packages/augur-ui/src/modules/common/validations.ts [hunk 1]
```diff
@@ -187,11 +188,29 @@ export function checkForUserInputFilled(inputs, endTimeFormatted) {
...
+        if (source && input.setEndTime <= source.setEndTime) {
+          return 'Date can not be before open date';
+        }
+      }
+      // day can not be a weekend
+      if (input.validationType === ValidationType.WEEKDAYONLY) {
+        if (input.setEndTime) {
+          const dayOfWeek = moment.unix(input.setEndTime).weekday()
+          if (dayOfWeek === SATURDAY_DAY_OF_WEEK || dayOfWeek === SUNDAY_DAY_OF_WEEK) {
+            return 'weekday is required';
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/common/validations.ts]: can this be capitalized?

Comment (bthaile) [reply packages/augur-ui/src/modules/common/validations.ts]: sure

**Review by phoebemirman — CHANGES_REQUESTED** [review]
Screen Shot 2019-12-30 at 10 36 33 AM

seems the event expiration should happen after the start date error isn't showing up here

**Review by phoebemirman — APPROVED** [review]

**Pull request merged**